### ISSUE TITLE
Need alternative for django-mailgun

### ISSUE THREAD

**Issue by pydanny** [pydanny/cookiecutter-django#531]
See https://github.com/BradWhittington/django-mailgun/issues/38

For what it's worth, in the past I tried to contact mailgun, but to no avail. I did it via connections as well as publicly. They don't seem to want to talk about providing support for Django. I'll try again, but I don't have my hopes up.

In the meantime, let's consider alternatives.

**Comment by djch** [comment]
For what it's worth, I started a new cookie cutter project the other day with Postmark, which is working well on a heroku-free-tier-addon.

* http://postmarkapp.com
* https://github.com/themartorana/python-postmark (this is not official and I dunno how well maintained it is honestly)
* https://devcenter.heroku.com/articles/postmark

I'm big fan of Wildbit.

**Comment by crdoconnor** [comment]
I've used sendgrid, which is okay and seems to be maintained: https://github.com/sendgrid/sendgrid-python/

However, I still don't really like using a library because it couples your application to a single transactional email provider. There's sometimes some value in the additional features this provides (e.g. to get some tracking features), but it makes it much to move providers.

This is especially valuable when something like [this](https://laravel-news.com/2016/02/mandrill-forcefully-requiring-mailchimp-accounts/) suddenly happens. It means you just need to change 3-4 lines of config in settings.py.

**Comment by pydanny** [comment]
Mailgun might be stepping up to correct the issue. I know @sigmavirus24 so I'm quite hopeful: https://github.com/BradWhittington/django-mailgun/issues/38#issuecomment-213226646

@djch, if mailgun doesn't work out, we'll check out Postmark.

@crdoconnor, we moved off of sendgrid, moving back is not an option. As for using a library, that's easier on us than having users set up Postfix in a container. In fact, my thought is that a savvy, competitive email-as-a-service company would make a Django package for working with their client one of their offered libraries. That's what I would do if I were them. 😉

**Comment by niklasae** [comment]
It might be worth keeping an eye on
https://github.com/anymail/django-anymail. It's based on the awesome Djrill
library, but supports a host of different providers.

**Comment by jayfk** [comment]
I don't get the point of `django-anymail`.

It's super easy to swap out django's mail backend. Why would you need a meta package to do that for you?

**Comment by pydanny** [comment]
@jayfk I think `django-anymail` is interesting because it provides a number of backends, not just one. It also appears to be pretty well supported.

**Comment by jayfk** [comment]
@pydanny after reading a bit of the code `django-anymail` is using, I'm in favor of switching. 

The code is well documented and a pleasure to read!

**Comment by jayfk** [comment]
Any reason not to switch @medmunds?

**Comment by medmunds** [comment]
@jayfk Anymail is still pretty young (and still marked "pre-alpha" in PyPI), but it benefits from a lot of lessons learned from years with Djrill. 

Basic sending functionality is stable, and I don't anticipate much (or any) churn around extensions like tagging and metadata. Event-tracking webhooks just went in, so might need some tweaks after real-world use. And a few other major features are still in the pipeline (ESP templates/merge/batch-send, and inbound email webhooks). But I don't know if any of those would matter for this project.

**Comment by pydanny** [comment]
I've decided that we'll move over to @medmunds's django-anymail. Here are some reasons:

1. Means Cookiecutter Django is much more mailhost agnostic. We can put in a default of mailgun, and over time we can support other options thanks to Cookiecutter's list item selection mechanism.
2. django-anymail, early stages or not, is being actively maintained. If Cookiecutter Django supports it, then the community for that project might grow, increasing the ability for it to support itself.
3. Much as I like @mailgun as a service, they aren't responsive enough to this particular issue of django-mailgun just not being supported. It's been 12 days since I opened this issue, and that's long enough.

I'll do the migration later this evening. After I'm done, if @medmunds could look over my work that would be great.
